### ISSUE TITLE
Script evaluated without error but finished with a false/empty top stack element

### ISSUE THREAD

**Issue by miguelmarco** [zcash/zcash#4903]
### Describe the issue

I am getting a  `16: mandatory-script-verify-flag-failed (Script evaluated without error but finished with a false/empty top stack element)` error when trying to broadcast a multisig transaction. I have tried to look at the decoded transaction, and as far as i can tell, it looks correct (although I am far from an expert, so maybe there is some subtlety that I am missing).

### Can you reliably reproduce the issue?

yes

#### If so, please list the steps to reproduce below:
1. run 
```
bin/zcash-cli sendrawtransaction 0400008085202f8901eda4b1c27eda0b3674e3af240425b416c1230110233a771639faf4eefcd59c8400000000fdfd0000483045022100b069540b7d32d47973ed2d26fc8d72d4d77d732fe79c4f3b6ba691e53c4e5a220220117e22999879f08e284de19d8b36153053495635b20e0c24d0624c5c022b5d10014730440220367517c047cc5cd069568be94f355b726e792687fa267b8072e32beb938669b9022020e3ff7e0386c4d3da411b3614ff451096d2630f4b20fb7c2dd849a26a4fcf8d014c69522103d33bd17c95ba8ab4ffc843543423689fd8eb41707bdf3ef4b67a32687c928c382102f7ec8e9d2c805916b2675cb2188b0be815ceec5316c92cec982042a3ded533e82102a63e95b747da60518c1a0349c16969f0d47f710b577bdf3ff9d42cd0effa753753aeffffffff01ea88e001000000001976a914fde7b3674f088aa87c369a453797d7602965480f88ac00000000000000000000000000000000000000
```

### Expected behaviour

I expected the stransaction to be accepted and broadcasted

### Actual behaviour + errors

I get the error message 

```error code: -26
error message:
16: mandatory-script-verify-flag-failed (Script evaluated without error but finished with a false/empty top stack element)
```

### The version of Zcash you were using:

`Zcash RPC client version v4.1.1

### Machine specs:
- OS name + version: Gentoo Linux
- CPU: AMD Ryzen 9 3900X 12-Core Processor
- RAM: 64GB
- Disk size: 1TB
- Disk Type (HD/SDD): HD
- Linux kernel version (uname -a): Linux neumann 5.3.9-gentoo #5 SMP PREEMPT Mon Nov 11 23:17:56 CET 2019 x86_64 AMD Ryzen 9 3900X 12-Core Processor AuthenticAMD GNU/Linux
- Compiler version (gcc --version): gcc (Gentoo 9.3.0-r1 p3) 9.3.0
- Linker version (ld -v): GNU ld (Gentoo 2.33.1 p2) 2.33.1
- Assembler version (as --version): Gentoo 2.33.1 p2

**Comment by str4d** [comment]
I looked at the transaction, and it does appear to be a standard 2-of-3 multisig. My guess is that the transaction signatures were created using an old consensus branch ID, that is no longer valid on the network (the consensus branch ID changes at each network upgrade, to enforce replay protection). How did you create the transaction?

**Comment by miguelmarco** [comment]
It was created by the openbazaar client. It was a disputed buy, so the moderator signed the 2 of 3 transaction.

From what you say, it seems that the openbazaar software didn't update to the corresponding consensus branch.

**Comment by str4d** [comment]
Yeah, if either of the two parties were using software that had an old consensus branch ID, their signature won't validate on the network. The transaction will need to be re-signed with the current consensus branch ID.
